File: GABRIEL_GELATTI_WEB_JANEIRO/src/test/java/stepsDefinitions/Hooks.java
package stepsDefinitions;

import io.cucumber.core.api.Scenario;
//import io.cucumber.core.api.Scenario;
import io.cucumber.java.After;
import io.cucumber.java.Before;
import pageObjects.LoginPage;

import static utils.Utils.*;

public class Hooks {
	
	@Before(value = "@logins")
	public void setupLogin() {
	    acessarSistema();
	}
	
	@Before(value = "not @logins")
	public void setup() {
	    acessarSistema();
		Na(LoginPage.class).realizarLogin("standard_user", "secret_sauce");
	}
	
	@After
	public void tearDown(Scenario cenario) {
		capturarTela(cenario);
		driver.quit();
	}
	
}

File: DesafioWebCSharp/DesafioWebCSharp/StepDefinitions/Hooks.cs
using TechTalk.SpecFlow;

namespace DesafioWebCSharp.StepDefinitions
{
    [Binding]
    public sealed class Hooks
    {
        private Utils _utils;

        public Hooks(Utils utils) => _utils = utils;

        [BeforeScenario]
        public void SetUp()
        {
            _utils.acessarSistema();
        }

        [AfterScenario]
        public void TearDown()
        {
            _utils.Driver.Quit();
        }
    }
}